plugin/evm/vm_test.go

```diff
@@ -92,97 +101,158 @@ var (
+	upgradeJSON string
+	configJSON  string
+}
+
+type testVM struct {
+	vm           *VM
+	toEngine     chan commonEng.Message
+	db           *prefixdb.Database
+	atomicMemory *atomic.Memory
+	appSender    *enginetest.Sender
+}
+
+func newVM(t *testing.T, config testVMConfig) *testVM {
+	ctx := utilstest.NewTestSnowContext(t, snowtest.CChainID)
+	fork := upgradetest.Latest
+	if config.fork != nil {
+		fork = *config.fork
+	}
+	ctx.NetworkUpgrades = upgradetest.GetConfig(fork)
+
+	if len(config.genesisJSON) == 0 {
+		config.genesisJSON = toGenesisJSON(forkToChainConfig[fork])
 	}
-	ctx := utils.TestSnowContext()
 
 	baseDB := memdb.New()
 
 	// initialize the atomic memory
 	atomicMemory := atomic.NewMemory(prefixdb.New([]byte{0}, baseDB))
 	ctx.SharedMemory = atomicMemory.NewSharedMemory(ctx.ChainID)
 
+	// NB: this lock is intentionally left locked when this function returns.
+	// The caller of this function is responsible for unlocking.
+	ctx.Lock.Lock()
+
 	issuer := make(chan commonEng.Message, 1)
 	prefixedDB := prefixdb.New([]byte{1}, baseDB)
-	return ctx, prefixedDB, []byte(genesisJSON), issuer, atomicMemory
-}
 
-// GenesisVM creates a VM instance with the genesis test bytes and returns
-// the channel use to send messages to the engine, the VM, database manager,
-// and sender.
-// If [genesisJSON] is empty, defaults to using [genesisJSONLatest]
-func GenesisVM(t *testing.T,
-	finishBootstrapping bool,
-	genesisJSON string,
-	configJSON string,
-	upgradeJSON string,
-) (
-	chan commonEng.Message,
-	*VM,
-	database.Database,
-	*enginetest.Sender,
-) {
 	vm := &VM{}
-	ctx, dbManager, genesisBytes, issuer, _ := setupGenesis(t, genesisJSON)
-	appSender := &enginetest.Sender{T: t}
-	appSender.CantSendAppGossip = true
-	appSender.SendAppGossipF = func(context.Context, commonEng.SendConfig, []byte) error { return nil }
+	appSender := &enginetest.Sender{
+		T:                 t,
+		CantSendAppGossip: true,
+		SendAppGossipF:    func(context.Context, commonEng.SendConfig, []byte) error { return nil },
+	}
 	err := vm.Initialize(
 		context.Background(),
 		ctx,
-		dbManager,
-		genesisBytes,
-		[]byte(upgradeJSON),
-		[]byte(configJSON),
+		prefixedDB,
+		[]byte(config.genesisJSON),
+		[]byte(config.upgradeJSON),
+		[]byte(config.configJSON),
 		issuer,
-		[]*commonEng.Fx{},
+		nil,
 		appSender,
 	)
-	require.NoError(t, err, "error initializing GenesisVM")
+	require.NoError(t, err, "error initializing vm")
 
-	if finishBootstrapping {
+	if !config.isSyncing {
 		require.NoError(t, vm.SetState(context.Background(), snow.Bootstrapping))
 		require.NoError(t, vm.SetState(context.Background(), snow.NormalOp))
 	}
 
-	return issuer, vm, dbManager, appSender
+	return &testVM{
+		vm:           vm,
+		toEngine:     issuer,
+		db:           prefixedDB,
+		atomicMemory: atomicMemory,
+		appSender:    appSender,
+	}
+}
+
+// setupGenesis sets up the genesis
+func setupGenesis(
+	t *testing.T,
+	fork upgradetest.Fork,
+	fallbackGenesisJSON string,
```

Comment:
I feel like we should not need that and just use genesisJson option in newVM

Reply:
Even if this would diverge further from coreth? Unless you're asking about the argument